pyproject.toml

```diff
@@ -0,0 +1,18 @@
+[project]
+name = "proxsuite"
+version = "0.0.1"
+description = "Quadratic Programming Solver for Robotics and beyond."
+readme = "README.md"
+requires-python = ">= 3.8"
+license = "BSD-2-Clause"
+
+
+[project.urls]
+homepage = "https://github.com/Simple-Robotics/proxsuite"
+repository = "https://github.com/Simple-Robotics/proxsuite.git"
+changelog = "https://github.com/Simple-Robotics/proxsuite/CHANGELOG.md"
+
+[build-system]
+requires = ["cmeel[build]"]
+build-backend = "cmeel.build"
+configure-args = ["-DCMAKE_CXX_STANDARD=17","-DCMAKE_CXX_COMPILER=gcc","-DBUILD_TESTING=OFF","-DBUILD_PYTHON_INTERFACE=ON","-DBUILD_WITH_VECTORIZATION_SUPPORT=ON","-DINSTALL_DOCUMENTATION=OFF"]
```

Comment:
```suggestion
configure-args = ["-DCMAKE_CXX_COMPILER=gcc","-DBUILD_TESTING:BOOL=OFF","-DBUILD_PYTHON_INTERFACE:BOOL=ON","-DBUILD_WITH_VECTORIZATION_SUPPORT:BOOL=ON","-DINSTALL_DOCUMENTATION:BOOL=OFF"]
```

Reply:
Could we use clang instead of gcc? Clang generates more efficient binaries.

Reply:
I tried it before and it did not work. I will try it again, for this I will trigger a pipeline here and see if it works with clang and without defining the c++ standard.

Reply:
after removing the standard here i get the following error in my pipeline:

```
      -- Default C++ standard:
      -- Minimal C++ standard upgraded to 17
      CMake Error at cmake-module/cxx-standard.cmake:101 (message):
        Unknown current C++ standard while trying to check for >= 17
```

(https://github.com/jcarpent/proxqp-unofficial/runs/7695050018?check_suite_focus=true#step:7:87)

why you think it is not ok to input it here, actually I can set it to whatever I want just it needs to be initialized I think.

Reply:
I think this is because the cmake module is not up-to-date. Have you rebased on latest devel branch?

Reply:
you are right. after checking out manually the latest master of the cmake-module it was fine. we should update this in the repo. now I have another problem with clang that I saw before and that made me use gcc. I think we have somehow problems with the standard library bc the manylinux docker that we are using are supposed to use the GCC (I think..)

```
      In file included from /__w/proxqp-unofficial/proxqp-unofficial/bindings/python/external/pybind11/include/pybind11/stl.h:22:
      /__w/proxqp-unofficial/proxqp-unofficial/3/envs/proxsuite/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/include/c++/10.4.0/valarray:1249:5: error: exception specification in declaration does not match previous declaration
          end(const valarray<_Tp>& __va) noexcept
          ^
      /__w/proxqp-unofficial/proxqp-unofficial/3/envs/proxsuite/bin/../lib/gcc/x86_64-conda-linux-gnu/10.4.0/../../../../x86_64-conda-linux-gnu/include/c++/10.4.0/bits/range_access.h:110:37: note: previous declaration is here
        template<typename _Tp> const _Tp* end(const valarray<_Tp>&);
```

https://github.com/jcarpent/proxqp-unofficial/runs/7695886810?check_suite_focus=true#step:8:2439

Reply:
OK. Keep then gcc. we will fix it later then.

Reply:
ok, I switched back to gcc, pipeline is passing.